### PR TITLE
Fix SQLCompatIT.testCursorFromOldNodeFailsOnNewNode

### LINKED ISSUE #85520: SQLCompatIT.
### CI Link

https://gradle-enterprise.elastic.co/s/vgdjkeiiridsk/tests/:x-pack:plugin:sql:qa:mixed-node:v8.2.0%23mixedClusterTest/org.elasticsearch.xpack.sql.qa.mixed_node.SqlCompatIT/testCursorFromOldNodeFailsOnNewNode?top-execution=1

### Repro line

./gradlew ':x-pack:plugin:sql:qa:mixed-node:v8.2.0#mixedClusterTest' -Dtests.class="org.elasticsearch.xpack.sql.qa.mixed_node.SqlCompatIT" -Dtests.method="testCursorFromOldNodeFailsOnNewNode" -Dtests.seed=FDCF684F324DE11D -Dtests.bwc=true -Dtests.locale=es-PR -Dtests.timezone=Asia/Baku -Druntime.java=17

### Does it reproduce?

Yes

### Applicable branches

master

### Failure history

_No response_

### Failure excerpt

```
java.lang.AssertionError:
  Expected: a string containing "Unsupported cursor version [8.2.0], expected [8.3.0]" |  
  but: was "method [POST], host [http://127.0.0.1:34941], URI [_sql], status line [HTTP/1.1 500 Internal Server Error] |  
  {"error":{"root_cause":[{"type":"sql_illegal_argument_exception","reason":"Unsupported cursor version [1.66.82], expected [8.3.0]"}],"type":"sql_illegal_argument_exception","reason":"Unsupported cursor version [1.66.82], expected [8.3.0]"},"status":500}"
```

### PR DESCRIPTION
**Pull request by Luegg**
resolves #85520

The failure was caused by the assumption that groupBy queries always return a scroll cursor that has been fixed in https://github.com/elastic/elasticsearch/pull/84356.

Because we will run into the same problem again with 8.2.1 this fix also needs to be backported to 8.2.

### CODE REVIEW

File: x-pack/plugin/sql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/sql/qa/mixed_node/SqlCompatIT.java [hunk 1]
```diff
@@ -174,11 +179,10 @@ private void assertCursorNotCompatibleAcrossVersions(Version version1, RestClien
         indexDocs();
 
         Request req = new Request("POST", "_sql");
-        // GROUP BY queries always return a cursor
-        req.setJsonEntity(sqlQueryEntityWithOptionalMode("SELECT int FROM test GROUP BY 1", bwcVersion));
+        req.setJsonEntity(sqlQueryEntityWithOptionalMode(Map.of("query", "SELECT int FROM test", "fetch_size", 1), bwcVersion));
         Map<String, Object> json = performRequestAndReadBodyAsJson(client1, req);
         String cursor = (String) json.get("cursor");
-        assertThat(cursor, Matchers.not(Matchers.emptyString()));
+        assertThat(cursor, Matchers.not(Matchers.emptyOrNullString()));
```
Comment (Luegg) [inline x-pack/plugin/sql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/sql/qa/mixed_node/SqlCompatIT.java:185]: the missing null check caused the `null` cursor to be propagated to the next request resulting in a `"null"` cursor.

**Review by astefan — APPROVED** [review]
LGTM. Thank you for the quick turnaround and fix.

**Review by luigidellaquila — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM. +1 for the fast resolution.

**Pull request merged**